test/test_mcmc.py

```diff
@@ -126,7 +126,8 @@ def model(data):
         12,  35,  17,  23,  17,   4,   2,  31,  30,  13,  27,   0,  39,  37,
         5,  14,  13,  22,
     ])
-    init_params, potential_fn, constrain_fn = initialize_model(random.PRNGKey(2), model, count_data)
+    init_params, potential_fn, constrain_fn = initialize_model(random.PRNGKey(2), model, count_data,
+                                                               initialize='prior')
     init_kernel, sample_kernel = hmc(potential_fn)
```

Comment:
@fehiepsi - I think there might be an issue with this test, specially with `assert max(tau_values) == 44`. No matter the number of warmup steps, I get ~ 70-71 with the uniform initialization strategy. I have changed it to `prior` above, but this is worth looking into.

Reply:
Looking at the plot at https://nbviewer.jupyter.org/github/CamDavidsonPilon/Probabilistic-Programming-and-Bayesian-Methods-for-Hackers/blob/master/Chapter1_Introduction/Ch1_Introduction_PyMC3.ipynb#Example:-Inferring-behaviour-from-text-message-data I think that we are just discovering another typical set of `tau`. The picture shows 2 highest modes: 1 is 44, 1 is 22. So 22 might be another solution of this problem. :D

Reply:
The strange thing is that with uniform initialization, I consistently see 70-71 as max value for tau and for prior initialization the max is around 44-45. I would have thought that increasing the number of samples / warmup would uncover all the modes, but that doesn't seem to be the case.

Reply:
I have reverted to using the prior distribution because the sampling seems to get stuck at one of the modes and give biased results. This is a great model to run with multiple chains and check for convergence. :)

Reply:
Yeah, this and your last regression model are the first HMC models which I see initialization plays an important role. :D :dancing_men:  It is good that we allow users provide custom initial values as their likes. Now, I understand why some users in forum and pyro ask for that feature.